src/v/http/tests/http_client_test.cc

```diff
@@ -804,3 +815,33 @@ SEASTAR_TEST_CASE(test_http_via_impostor_no_content_length) {
           });
     });
 }
+
+SEASTAR_TEST_CASE(test_http_cancel_reconnect) {
```

Comment:
there is a `SEASTAR_THREAD_TEST_CASE` which doesn't require you to do `ss::async` every time.